datafusion/physical-plan/src/joins/cross_join.rs

```diff
@@ -347,13 +348,14 @@ fn build_batch(
         })
         .collect::<Result<Vec<_>>>()?;
 
-    RecordBatch::try_new(
+    RecordBatch::try_new_with_options(
```

Comment:
I wonder if we have more operators with empty inputs that use `RecordBatch::try_new` that should move to use `RecordBatch::try_new_with_options`

Reply:
I have thought about this too, but I didn't come up with any more tests😥.

Reply:
I think it is fine to merge as is and see if we get any regressions (hopefully not), those should be easy to fix in a similar manner.